compose/compose.go

```diff
@@ -116,6 +136,36 @@ func (c *Compose) MustKill() {
 	}
 }
 
+// GetPublicIPAddressForService returns the IPAddress of a service
+func (c *Compose) GetPublicIPAddressForService(serviceName string) (bool, string) {
```

Comment:
Sorry I'm still not sure I'm clear on this one - my understanding is that if you want to access a container from a test (i.e. from outside the virtual Docker network) your only option is to ask Docker to bind an exposed port from a container to a port on the host network, because the internal IP address of the container won't be reachable from the host otherwise.

You can do that using the "ports" section in the compose file. You won't be able to specify the port in the "<host>:<container>" format as the second container will fail to start because the port on the host is already taken. If you only specify the "<container>" port, Docker will automatically allocate a free port on the host network, and you can find out which one it is using `MustGetFirstPublicPort`.

Maybe I'm missing something though - is there a way to route from the host network to an IP address in the virtual Docker network?

Note that if you are instead trying to connect to a container from another one, you can just specify the dependency in the "links" question and use the container name as hostname, as the virtual Docker network has a DNS server which will resolve it to the internal IP address of the container.